tests/components/rainbird/test_init.py

```diff
@@ -264,18 +282,153 @@ async def test_fix_entity_unique_ids(
+
+    device_entry = device_registry.async_get_device(
+        {(DOMAIN, expected_device_identifier)}
+    )
+    assert device_entry
+    assert device_entry.identifiers == {(DOMAIN, expected_device_identifier)}
+
+
+@pytest.mark.parametrize(
+    (
+        "entry1_updates",
+        "entry2_updates",
+        "expected_device_name",
+        "expected_disabled_by",
+    ),
+    [
+        ({}, {}, None, None),
+        (
+            {
+                "name_by_user": "Front Sprinkler",
+            },
+            {},
+            "Front Sprinkler",
+            None,
+        ),
+        (
+            {},
+            {
+                "name_by_user": "Front Sprinkler",
+            },
+            "Front Sprinkler",
+            None,
+        ),
+        (
+            {
+                "name_by_user": "Sprinkler 1",
+            },
+            {
+                "name_by_user": "Sprinkler 2",
+            },
+            "Sprinkler 2",
+            None,
+        ),
+        (
+            {
+                "disabled_by": dr.DeviceEntryDisabler.USER,
+            },
+            {},
+            None,
+            None,
+        ),
+        (
+            {},
+            {
+                "disabled_by": dr.DeviceEntryDisabler.USER,
+            },
+            None,
+            None,
+        ),
+        (
+            {
+                "disabled_by": dr.DeviceEntryDisabler.USER,
+            },
+            {
+                "disabled_by": dr.DeviceEntryDisabler.USER,
+            },
+            None,
+            dr.DeviceEntryDisabler.USER,
+        ),
+    ],
+    ids=[
+        "duplicates",
+        "prefer-old-name",
+        "prefer-new-name",
+        "both-names-prefers-new",
+        "old-disabled-prefer-new",
+        "new-disabled-prefer-old",
+        "both-disabled",
+    ],
+)
+async def test_fix_duplicate_device_ids(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    entry1_updates: dict[str, Any],
+    entry2_updates: dict[str, Any],
+    expected_device_name: str | None,
+    expected_disabled_by: dr.DeviceEntryDisabler | None,
+) -> None:
+    """Test fixing duplicate device ids."""
+
+    entry1 = device_registry.async_get_or_create(
+        identifiers={(DOMAIN, str(SERIAL_NUMBER))},
+        config_entry_id=config_entry.entry_id,
+        serial_number=config_entry.data["serial_number"],
+    )
+    device_registry.async_update_device(entry1.id, **entry1_updates)
+
+    entry2 = device_registry.async_get_or_create(
+        identifiers={(DOMAIN, MAC_ADDRESS_UNIQUE_ID)},
+        config_entry_id=config_entry.entry_id,
+        serial_number=config_entry.data["serial_number"],
+    )
+    device_registry.async_update_device(entry2.id, **entry2_updates)
+
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry.entry_id
+    )
+    assert len(device_entries) == 2
+
+    await config_entry.async_setup(hass)
```

Comment:
Please use `hass.config_entries.async_setup` instead.

Reply:
Ah whoops. I started using this in a bunch of tests not realizing i'm the only one doing this.

https://github.com/search?q=repo%3Ahome-assistant%2Fcore%20%22config_entry.async_setup%22&type=code